truststore.py

```diff
@@ -380,7 +381,52 @@ def _verify_peercerts_impl(cert_chain: List[bytes], server_hostname=None) -> Non
 # ===== Windows =====
 
 elif platform.system() == "Windows":
-    raise ImportError("Windows is not yet supported")
+
+    from ctypes import POINTER, Structure, WinDLL, c_char_p, c_ulong, c_void_p
+    from ctypes.wintypes import (
+        BOOL,
+        BYTE,
+        CHAR,
+        DWORD,
+        HWND,
+        LONG,
+        LPCSTR,
+        LPCWSTR,
+        LPFILETIME,
+        LPSTR,
+        LPWSTR,
+        ULONG,
+        USHORT,
+    )
+
+    WINTRUST = WinDLL("wintrust.dll")
+    CRYPT32 = WinDLL("crypt32.dll")
+
+    HTTPSPROV_ACTION = WINTRUST.HTTPSPROV_ACTION
+    WTD_UI_NONE = WINTRUST.WTD_UI_NONE
+    WTD_REVOCATION_CHECK_CHAIN = WINTRUST.WTD_REVOCATION_CHECK_CHAIN
+    WTD_CHOICE_CERT = WINTRUST.WTD_CHOICE_CERT
+    AUTHTYPE_SERVER = WINTRUST.AUTHTYPE_SERVER
+    X509_ASN_ENCODING = WINTRUST.X509_ASN_ENCODING
+    PKCS_7_ASN_ENCODING = WINTRUST.PKCS_7_ASN_ENCODING
+
+    WinVerifyTrust = WINTRUST.WinVerifyTrust
+    WinVerifyTrust.argtypes = [HWND, c_void_p, LPVOID]
+    WinVerifyTrust.restype = LONG
```

Comment:
https://docs.microsoft.com/en-us/windows/win32/api/wintrust/nf-wintrust-winverifytrust says "For certificate verification, use the [CertGetCertificateChain](https://docs.microsoft.com/en-us/windows/desktop/api/wincrypt/nf-wincrypt-certgetcertificatechain) and [CertVerifyCertificateChainPolicy](https://docs.microsoft.com/en-us/windows/desktop/api/wincrypt/nf-wincrypt-certverifycertificatechainpolicy) functions," which makes me 2nd-guess whether WinVerifyTrust is the right API to use.

Reply:
It looks like possibly what we should do is:
    - Create an in-memory certificate store using [CertOpenStore](https://docs.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certopenstore) with lpszStoreProvider set to CERT_STORE_PROV_MEMORY
    - Add any intermediate certs from our peer to it using [CertAddEncodedCertificateToStore](https://docs.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certaddencodedcertificatetostore)
    - Get a cert chain using [CertGetCertificateChain](CertGetCertificateChain) with hAdditionalStore set to the in-memory store
    - Verify the chain using [CertVerifyCertificateChainPolicy](https://docs.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certverifycertificatechainpolicy)